### PR TITLE
Improve analytics partition in Puppet demo

### PR DESCRIPTION
**Pull request by erychagov**
Implemented analytics delegate. Added UI tests for Analytics.

### CODE REVIEW

**Comment by TroubleMakerBen** [comment]
Have you tried `reloadRowsAtIndexPaths:`?

File: Puppet/Puppet/AppDelegate.m [hunk 1]
```diff
@@ -13,8 +14,11 @@
 #import "MobileCenterDistribute.h"
 #import "MobileCenterPush.h"
 #import "MSAlertController.h"
+#import "MSAnalyticsDelegate.h"
+#import "MSAnalyticsInternal.h"
+#import "MSEventLog.h"
 
-@interface AppDelegate () <MSCrashesDelegate, MSDistributeDelegate, MSPushDelegate>
+@interface AppDelegate () <MSCrashesDelegate, MSDistributeDelegate, MSPushDelegate, MSAnalyticsDelegate>
```
Comment (jaelim-ms) [inline Puppet/Puppet/AppDelegate.m]: Any reason of using MSAnalyticsDelegate? It is Xamarin only delegate and eventually we will remove or at least revise this one. This delegate is not a public delegate that can be used by applications.

**Comment by erychagov** [comment]
@TroubleMakerBen 

> Have you tried reloadRowsAtIndexPaths:?

Yes, I use it in the current realisation.

File: Puppet/PuppetUITests/AnalyticsUITests.swift [hunk 2]
```diff
@@ -4,6 +4,10 @@ class AnalyticsUITests: XCTestCase {
   private var app : XCUIApplication?;
   private let AnalyticsCellIndex : UInt = 0;
 
+  private let kDidSentEventText : String = "Sent event occurred";
+  private let kDidFailedToSendEventText : String = "Failed to send event occurred";
+  private let kDidSendingEventText : String = "Sending event occurred";
```
Comment (jaelim-ms) [inline Puppet/PuppetUITests/AnalyticsUITests.swift]: Shouldn't it be *Event sent*, *Failed to send event* and *Sending event*?

Comment (erychagov) [inline Puppet/PuppetUITests/AnalyticsUITests.swift]: I use [this file](https://github.com/Microsoft/mobile-center-sdk-dotnet/blob/develop/Tests/Contoso.Forms.Test.Strings/Strings.cs#L118) as example

Comment (jaelim-ms) [inline Puppet/PuppetUITests/AnalyticsUITests.swift]: Oh.. OK. I see.

**Pull request merged**